### PR TITLE
Use interpolated strings 1

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary

Replace `string.Format()` with new `string.Create((IFormatProvider? provider, [InterpolatedStringHandlerArgument(nameof(provider))] ref DefaultInterpolatedStringHandler handler)`.

## PR Context
#18974 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.Management.UI.Internal/HelpWindow/HelpParagraphBuilder.cs [hunk 1]
```diff
@@ -276,7 +276,7 @@ private static string AddIndent(string str, string indentString)
             foreach (string line in lines)
             {
                 // Indentation is not localized
-                returnValue.AppendFormat("{0}{1}\r\n", indentString, line);
+                returnValue.AppendFormat($"{indentString}{line}\r\n");
```
Comment (iSazonov) [inline src/Microsoft.Management.UI.Internal/HelpWindow/HelpParagraphBuilder.cs:279]:
```suggestion
                returnValue.Append($"{indentString}{line}\r\n");
```

File: src/Microsoft.Management.UI.Internal/ManagementList/ManagementList/Innerlist.cs [hunk 2]
```diff
@@ -596,7 +596,7 @@ private string GetClipboardTextLineForSelectedItem(object value)
                     propertyValue = string.Empty;
                 }
 
-                entryText.AppendFormat(CultureInfo.CurrentCulture, "{0}\t", propertyValue);
+                entryText.AppendFormat(CultureInfo.CurrentCulture, $"{propertyValue}\t");
```
Comment (iSazonov) [inline src/Microsoft.Management.UI.Internal/ManagementList/ManagementList/Innerlist.cs:599]:
```suggestion
                entryText.Append(CultureInfo.CurrentCulture, $"{propertyValue}\t");
```

File: src/Microsoft.Management.UI.Internal/ShowCommand/Controls/ParameterSetControl.xaml.cs [hunk 3]
```diff
@@ -164,7 +164,7 @@ private static MultipleSelectionControl CreateMultiSelectComboControl(ParameterV
             multiControls.comboxParameter.SetBinding(ComboBox.TextProperty, valueBinding);
 
             // Add AutomationProperties.AutomationId for Ui Automation test.
-            multiControls.SetValue(System.Windows.Automation.AutomationProperties.AutomationIdProperty, string.Format("combox{0}", parameterViewModel.Name));
+            multiControls.SetValue(System.Windows.Automation.AutomationProperties.AutomationIdProperty, string.Format($"combox{parameterViewModel.Name}"));
```
Comment (iSazonov) [inline src/Microsoft.Management.UI.Internal/ShowCommand/Controls/ParameterSetControl.xaml.cs:167]:
```suggestion
            multiControls.SetValue(System.Windows.Automation.AutomationProperties.AutomationIdProperty, string.Create($"combox{parameterViewModel.Name}"));
```

File: src/Microsoft.Management.UI.Internal/ShowCommand/ViewModel/CommandViewModel.cs [hunk 4]
```diff
@@ -431,7 +431,7 @@ public string GetScript()
 
             if (commandName.Contains(' '))
             {
-                builder.AppendFormat("& \"{0}\"", commandName);
+                builder.AppendFormat($"& \"{commandName}\"");
```
Comment (iSazonov) [inline src/Microsoft.Management.UI.Internal/ShowCommand/ViewModel/CommandViewModel.cs:434]:
```suggestion
                builder.Append($"& \"{commandName}\"");
```

File: src/Microsoft.Management.UI.Internal/ShowCommand/ViewModel/ParameterSetViewModel.cs [hunk 5]
```diff
@@ -138,7 +138,7 @@ public string GetScript()
                 {
                     if (((bool?)parameter.Value) == true)
                     {
-                        builder.AppendFormat("-{0} ", parameter.Name);
+                        builder.AppendFormat($"-{parameter.Name} ");
```
Comment (iSazonov) [inline src/Microsoft.Management.UI.Internal/ShowCommand/ViewModel/ParameterSetViewModel.cs:141]:
```suggestion
                        builder.Append($"-{parameter.Name} ");
```

File: src/Microsoft.Management.UI.Internal/ShowCommand/ViewModel/ParameterSetViewModel.cs [hunk 6]
```diff
@@ -166,7 +166,7 @@ public string GetScript()
                     parameterValueString = ParameterSetViewModel.GetDelimitedParameter(parameterValueString, "(", ")");
                 }
 
-                builder.AppendFormat("-{0} {1} ", parameter.Name, parameterValueString);
+                builder.AppendFormat($"-{parameter.Name} {parameterValueString} ");
```
Comment (iSazonov) [inline src/Microsoft.Management.UI.Internal/ShowCommand/ViewModel/ParameterSetViewModel.cs:169]:
```suggestion
                builder.Append($"-{parameter.Name} {parameterValueString} ");
```

**Comment by iSazonov** [comment]
> D:\a\1\s\src\System.Management.Automation\SourceGenerators\PSVersionInfoGenerator\PSVersionInfoGenerator.cs(14,18): warning RS1036: 'SMA.PSVersionInfoGenerator': A project containing analyzers or source generators should specify the property '<EnforceExtendedAnalyzerRules>true</EnforceExtendedAnalyzerRules>' [D:\a\1\s\src\System.Management.Automation\SourceGenerators\PSVersionInfoGenerator\PSVersionInfoGenerator.csproj]

I guess this asks to add `<EnforceExtendedAnalyzerRules>true</EnforceExtendedAnalyzerRules>` in PSVersionInfoGenerator.csproj 

> D:\a\1\s\src\Microsoft.Management.UI.Internal\ShowCommand\Controls\ParameterSetControl.xaml.cs(164,112): error CS1501: No overload for method 'Create' takes 1 arguments [D:\a\1\s\src\Microsoft.Management.UI.Internal\Microsoft.PowerShell.GraphicalHost_l0ynpe1u_wpftmp.csproj]

Please fix.

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.Management.Infrastructure.CimCmdlets/CimSessionProxy.cs [hunk 7]
```diff
@@ -875,7 +875,7 @@ internal void WriteOperationStartMessage(string operation, Hashtable parameterLi
                         parameters.Append(',');
                     }
 
-                    parameters.Append(string.Format(CultureInfo.CurrentUICulture, @"'{0}' = {1}", key, parameterList[key]));
+                    parameters.AppendFormat(CultureInfo.CurrentUICulture, $@"'{key}' = {parameterList[key]}");
```
Comment (iSazonov) [inline src/Microsoft.Management.Infrastructure.CimCmdlets/CimSessionProxy.cs:878]:
```suggestion
                    parameters.Append(CultureInfo.CurrentUICulture, $@"'{key}' = {parameterList[key]}");
```

File: src/Microsoft.Management.UI.Internal/ShowCommand/Controls/MultipleSelectionControl.xaml.cs [hunk 8]
```diff
@@ -43,7 +43,7 @@ private void ButtonBrowse_Click(object sender, RoutedEventArgs e)
 
             foreach (object selectedItem in multipleSelectionDialog.listboxParameter.SelectedItems)
             {
-                newComboText.AppendFormat(CultureInfo.InvariantCulture, "{0},", selectedItem.ToString());
+                newComboText.AppendFormat(CultureInfo.InvariantCulture, $"{selectedItem},");
```
Comment (iSazonov) [inline src/Microsoft.Management.UI.Internal/ShowCommand/Controls/MultipleSelectionControl.xaml.cs:46]:
```suggestion
                newComboText.Append(CultureInfo.InvariantCulture, $"{selectedItem},");
```

**Comment by CarloToso** [comment]
@iSazonov Do we still need to fix something to merge this PR?

**Pull request merged**